### PR TITLE
Remove shrapnel button

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Added "remove shrapnel" button to surgery UI, it's not visible when there is no shrapnel to remove.
Requires hemostat / wirecutters and skill check that is somewhat harder than diagnostic check.

image

## Why It's Good For The Game

Healing people packed with shrapnel is too much pain.

## Changelog
:cl:
add: shrapnel removal button in surgery UI
/:cl:

### CODE REVIEW

**Comment by SirRichardFrancis** [comment]
image
image

**Review by MLGTASTICa — COMMENTED** [review]
Consider making it loop instead of removing all shrapnel at once ? it lets people far too easily negate grenade shrapnel if they get a huge amount of it and do combat surgery with a chair or are a FBP.

**Comment by SirRichardFrancis** [comment]
> Consider making it loop instead of removing all shrapnel at once ? it lets people far too easily negate grenade shrapnel if they get a huge amount of it and do combat surgery with a chair or are a FBP.

That's the whole point. We have ridiculous amounts of shrapnel generated by grenades and some guns, removing it shouldn't be hour-long cbt.

**Comment by TheShown911** [comment]
If it loops it won't be CBT, as there would be only one button that is pressed, but it would be an hour long.

**Comment by SirRichardFrancis** [comment]
> but it would be an hour long.

Is that... good? There is already a hard skill check in place and it's not instant.

File: code/modules/surgery/surgery_ui.dm [hunk 1]
```diff
@@ -156,4 +157,35 @@
...
 
-			return TRUE
\ No newline at end of file
+			return TRUE
+
+		if("remove_shrapnel")
+			if(istype(usr, /mob/living))
+				var/mob/living/user = usr
+				var/target_stat = BP_IS_ROBOTIC(src) ? STAT_MEC : STAT_BIO
+				var/removal_time = 70 * usr.stats.getMult(target_stat, STAT_LEVEL_PROF)
```
Comment (hyperioo) [inline code/modules/surgery/surgery_ui.dm:166]:
```suggestion
				var/removal_time = 70 * user.stats.getMult(target_stat, STAT_LEVEL_PROF)
```

File: code/modules/surgery/surgery_ui.dm [hunk 2]
```diff
@@ -156,4 +157,35 @@
...
+				to_chat(user, SPAN_NOTICE("You start removing shrapnel from [get_surgery_name()]."))
+
+				var/wait
+				if(ismob(target))
+					wait = do_mob(user, target, removal_time)
+				else
+					wait = do_after(user, removal_time, target, needhand = FALSE)
+
+				if(wait)
+					if(prob(100 - FAILCHANCE_NORMAL + usr.stats.getStat(target_stat)))
```
Comment (hyperioo) [inline code/modules/surgery/surgery_ui.dm:183]:
```suggestion
					if(prob(100 - FAILCHANCE_NORMAL + user.stats.getStat(target_stat)))
```

**Review by hyperioo — APPROVED** [review]
Should work with `usr` but `user` would be better for consistency

**Pull request merged**